test/studies/prk/stencil/stencil-fast.chpl

```diff
@@ -1,238 +1,257 @@
-
-  // Choice of distribution / parallelism
-  const blockDist = new dmap(new Block(localDom)),
-      stencilDist = new dmap(new Stencil(innerLocalDom, fluff=(R,R))),
-           noDist = new dmap(new DefaultDist()),
-         replDist = new dmap(new ReplicatedDist());
-
-  const Dist =  if useBlockDist then blockDist
-                else if useStencilDist then stencilDist
-                else noDist;
+/* Parallel Research Kernel - Stencil */
 
-  //const weightDist = if (useBlockDist || useStencilDist) then replDist
-  //                 else noDist;
-
-  // Domains
-  const Dom =      localDom dmapped Dist,
-        innerDom = innerLocalDom dmapped Dist;
-
-  var tiledDom = {R.. # order-2*R by tileSize, R.. # order-2*R by tileSize};
-
-  // Arrays (initialized to zeros)
-  var input, output:  [Dom] dtype = 0.0;
-
-
-  // Tuple of tuples
-  pragma "locale private"
-  var weight: Wsize*(Wsize*(dtype));
+//
+// Process and test input configs
+//
+if (iterations < 1) {
+  writeln("ERROR: iterations must be >= 1: ", iterations);
+  exit(1);
+}
+if (order < 1) {
+  writeln("ERROR: Matrix Order must be greater than 0 : ", order);
+  exit(1);
+}
+if (R < 1) {
+  writeln("ERROR: Stencil radius ", R, " should be positive");
+  exit(1);
+}
+if (2*R + 1 > order) {
+  writeln("ERROR: Stencil radius ", R, " exceeds grid size ", order);
+  exit(1);
+}
 
-  // Create local copy of weight on each Locale
-  for loc in Locales do on loc {
-    for i in 1..R {
-      const element : dtype = 1 / (2*i*R) : dtype;
-      weight[R1][R1+i]  =  element;
-      weight[R1+i][R1]  =  element;
-      weight[R1-i][R1] = -element;
-      weight[R1][R1-i] = -element;
-    }
+/* If tiling is disabled, ensure tiledDom can still be created */
+if (!tiling) then tileSize = 1;
+
+/* Domain over entire Input/Output matrices */
+const localDom = {0.. # order, 0.. # order},
+/* Domain over entire 'active' region, where Output will be updated */
+ innerLocalDom = localDom.expand(-R),
+/* Strided domain over 'active' region - possibly a cleaner way to write...*/
+tiledLocalDom = {R.. # order-2*R by tileSize, R.. # order-2*R by tileSize};
+
+/* Flavors of parallelism, by distribution */
+const blockDist = new Block(localDom),
+    stencilDist = new Stencil(innerLocalDom, fluff=(R,R)),
+         noDist = new DefaultDist();
+
+/* Set distribution based on configs */
+const Dist =  if useBlockDist then blockDist
+              else if useStencilDist then stencilDist
+              else noDist;
+
+/* Map domains to selected distribution */
+const Dom = localDom dmapped new dmap(Dist),
+ innerDom = innerLocalDom dmapped new dmap(Dist),
+ tiledDom = tiledLocalDom dmapped new dmap(Dist);
+
+/* Input and Output matrices represented as arrays over a 2D domain */
+var input, output:  [Dom] dtype = 0.0;
+
+/* Pragma ensures each locale stores a local copy of weight matrix */
+pragma "locale private"
+/* Weight matrix represented as tuple of tuples*/
+var weight: Wsize*(Wsize*(dtype));
```

Comment:
I think it would be simpler to just abandon the locale-private approach in favor of forall-intents. I think it's cleaner and allows you to keep the main function.